pre_release.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+usage() { echo "Usage: $0 -t <tag>" 1>&2; exit 1; }
+
+while getopts ":t:" o; do
+    case "${o}" in
+        t)
+            tag=${OPTARG}
+            ;;
+        *)
+            usage
+            ;;
+    esac
+done
+if [ ! -z "${t}" ]; then
+    usage
+fi
+
+#validate tag
```

Comment:
nit: give some example on semver in comment?

Reply:
I think we're a little bit reinventing the wheel here, since most of this - including `semver` convention is documented here: https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/managing-releases-in-a-repository - why not adopt this process?

Reply:
Please refer to my comment on managing changelog.md.